src/lib/state/settings.ts

```diff
@@ -17,7 +20,7 @@ interface Settings {
 
 const initialSettings: Settings = {
   maxSlippage: 'auto',
-  transactionTtl: TRANSACTION_TTL_DEFAULT,
+  transactionTtl: undefined,
```

Comment:
If we pre-populate this with a default then it is impossible for us to determine if the value was set by default or by a user without including additional state. Because we default to the default value, keeping this as undefined can inform our UX as to whether the input should be empty (ie use a placeholder) or hold a user-inputted value.

Reply:
i think we return an empty swap callback if the deadline is undefined - but maybe should update within that hook instead 

see how deadline is grabbed here https://github.com/Uniswap/interface/blob/c82b4fae6479dd8aa62a8a3f8859d0ea32d75c85/src/lib/hooks/useTransactionDeadline.ts#L20

and passed into useSwapCallback which ends up using it here: https://github.com/Uniswap/interface/blob/c82b4fae6479dd8aa62a8a3f8859d0ea32d75c85/src/hooks/useSwapCallArguments.tsx#L49

Reply:
Updated that hook to return default if not specified in atom.